test/unit/org/apache/cassandra/service/PartitionDenylistTest.java

```diff
@@ -0,0 +1,331 @@
+        process("SELECT * FROM " + ks_cql + ".foofoo", ConsistencyLevel.ONE);
+    }
+
+    /**
+     * Want to make sure we don't throw anything or explode when people try to remove a key that's not there
+     */
+    @Test
+    public void testRemoveMissingIsGraceful() throws InterruptedException
+    {
+        confirmDenied("bbb", "ccc");
+        Assert.assertTrue(removeDenylist("" + ks_cql + "", "foofoo", "bbb:ccc"));
+
+        // We expect this to silently not find and succeed at *trying* to remove it
+        Assert.assertTrue(removeDenylist("" + ks_cql + "", "foofoo", "bbb:ccc"));
+
+        confirmAllowed("bbb", "ccc");
+    }
+
+    /**
+     * We need to confirm that the entire cache is reloaded rather than being an additive change; we don't want keys to
+     * persist after their removal and reload from CQL.
+     */
+    @Test
+    public void testRemoveWorksOnReload() throws InterruptedException
+    {
+        denyAllKeys();
+
+        confirmDenied("aaa", "bbb");
+        confirmDenied("eee", "fff");
+        confirmDenied("iii", "jjj");
+
+        // poke a hole in the middle and reload
+        removeDenylist("" + ks_cql + "", "foofoo", "eee:fff");
+
+        confirmAllowed("eee", "fff");
+        confirmDenied("aaa", "bbb");
+        confirmDenied("iii", "jjj");
+    }
+
+    /**
+     * We go through a few steps here:
+     *  1) Add more keys than we're allowed
+     *  2) Confirm that the overflow keys are *not* denied
+     *  3) Raise the allowable limit
+     *  4) Confirm that the overflow keys are now denied (and no longer really "overflow" for that matter)
+     */
+    @Test
+    public void testShrinkAndGrow() throws InterruptedException
+    {
+        StorageProxy.instance.setMaxDenylistKeysPerTable(5);
+        StorageProxy.instance.setMaxDenylistKeysTotal(5);
+        denyAllKeys();
+
+        // Confirm overflowed keys are allowed; first come first served
+        confirmDenied("aaa", "bbb");
+        confirmAllowed("iii", "jjj");
+
+        // Now we raise the limit back up and do nothing else and confirm it's blocked
+        StorageProxy.instance.setMaxDenylistKeysPerTable(1000);
+        StorageProxy.instance.setMaxDenylistKeysTotal(1000);
+        confirmDenied("aaa", "bbb");
+        confirmDenied("iii", "jjj");
+
+        StorageProxy.instance.setMaxDenylistKeysPerTable(5);
+        StorageProxy.instance.setMaxDenylistKeysTotal(5);
+        confirmAllowed("iii", "jjj");
+
+        // Now, we remove the denylist entries for our first 5, drop the limit back down, and confirm those overflowed keys now block
+        removeDenylist("" + ks_cql + "", "foofoo", "aaa:bbb");
+        removeDenylist("" + ks_cql + "", "foofoo", "bbb:ccc");
+        removeDenylist("" + ks_cql + "", "foofoo", "ccc:ddd");
+        removeDenylist("" + ks_cql + "", "foofoo", "ddd:eee");
+        removeDenylist("" + ks_cql + "", "foofoo", "eee:fff");
+        confirmDenied("iii", "jjj");
```

Comment:
I got the test idea, but I'm not clear why it works (if it is). I assume you cache 5 keys here and then remove them, but the cache is not reloaded, so theoretically the key still should not be loaded.

Does it work because all 10 keys were loaded in the first place despite `StorageProxy.instance.setMaxDenylistKeysPerTable(5);`?